src/auth.ts

```diff
@@ -0,0 +1,61 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
+
+export const M2_DIR = '.m2';
+export const SETTINGS_FILE = 'settings.xml';
+
+export async function configAuthentication(
+  id: string,
+  username: string,
+  password: string
+) {
+  if (id && username && password) {
+    console.log(
+      `creating ${SETTINGS_FILE} with server-id: ${id}, username: ${username}, and a password`
+    );
+    const directory: string = path.join(os.homedir(), M2_DIR);
+    await io.mkdirP(directory);
+    core.debug(`created directory ${directory}`);
+    await write(directory, generate(id, username, password));
+  } else {
+    core.debug(
+      `no ${SETTINGS_FILE} without server-id: ${id}, username: ${username}, and a password`
+    );
+  }
+}
+
+// only exported for testing purposes
+export function generate(id: string, username: string, password: string) {
```

Comment:
You're not sanitizing XML control characters here.
You need to replace XML control characters with their 'safe' counterparts.

https://stackoverflow.com/a/7918944/12344388

Reply:
Good catch! fixed this in 551e2a2